File: python/2582.py
class Solution:
    def passThePillow(self, n: int, time: int) -> int:
        # if n == 1:
        #     return 1
        # location = time % (2 * n - 2) + 1  # 1-based
        # return location if location <= n else 2 * n - location
        return 1 if n == 1 else (time % (2 * n - 2) + 1 if time % (2 * n - 2) + 1 <= n else 2 * n - time % (2 * n - 2) - 1)
        
if __name__ == '__main__':
    s = Solution()
    n = int(input())
    time = int(input())
    print(s.passThePillow(n, time))

File: cpp/2582.cpp
#include<iostream>

using namespace std;

class Solution {
public:
    int passThePillow(int n, int time) {
        return n == 1 ? 1 : time % (2 * n - 2) + 1 <= n ? time % (2 * n - 2) + 1 : 2 * n - time % (2 * n - 2) - 1;
    }
};

int main(){
    Solution s;
    int n, time;
    cin >> n >> time;
    cout << s.passThePillow(n, time) << endl;
    return 0;
}